File: app-server/src/main/java/com/example/app/repository/UserRepository.java
package com.example.app.repository;

import com.example.app.model.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface UserRepository extends JpaRepository<User, String> {
    User findByUserName(String userName);
    User findByEmail(String email);
}

File: 2024-05-22/MyActivity2/app/src/main/java/com/example/myactivity/data/repository/UserRepository.kt
package com.example.myactivity.data.repository

import com.example.myactivity.data.network.ApiService
import com.example.myactivity.data.model.User

class UserRepository(private val apiService: ApiService) {

    suspend fun login(email: String, password: String): Boolean {
        return apiService.login(email, password)
    }

    suspend fun register(user: User): Boolean {
        return apiService.register(user)
    }

    suspend fun naverLogin(accessToken: String): Boolean {
        return apiService.naverLogin(accessToken)
    }
}